src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -348,20 +347,28 @@ private JObject GetVerificationResult(UInt160 scriptHash, ContractParameter[] ar
             {
                 throw new RpcException(-100, "Unknown contract");
             }
-            var methodName = "verify";
+            var md = contract.Manifest.Abi.GetMethod("verify", -1);
+            if (md is null)
+                throw new RpcException(-101, $"The smart contract {contract.Hash} haven't got verify method.");
+            if (md.ReturnType != ContractParameterType.Boolean)
+                throw new RpcException(-102, "The verify method doesn't return boolean value.");
 
-            Transaction tx = signers == null ? null : new Transaction
+            Transaction tx = new Transaction
             {
-                Signers = signers.GetSigners(),
-                Attributes = Array.Empty<TransactionAttribute>()
+                Signers = signers == null ? new Signer[] { new() { Account = scriptHash } } : signers.GetSigners(),
+                Attributes = Array.Empty<TransactionAttribute>(),
+                Script = new[] { (byte)OpCode.RET }
```

Comment:
And the last issue: at the current moment, the transaction is completely missing witnesses. I would suggest to pass witnesses along with signers via RPC request parameters using the following structure:
```
{
    "jsonrpc": "2.0",
    "method": "invokecontractverify",
    "params": [
        "0x31f714e04766139c8f3705f5ce94518e1411ec07",
        [ ],
        [
            {
                "account": "NiRqSd5MtRZT5yUhgWd7oG11brkDG76Jim",
                "scopes": "CalledByEntry",
                "invocation": "DEDr2gA/8T/wxQvgOZVfCdkbj6uGrprkDgJvpOJCcbl+tvlKZkZytCZEWm6NoZhJyIlEI3VQSLtU3AHuJfShAT5L",
                "verification": "DCEDAS1H52IQrsc745qz0YbgpA/o2Gv6PU+r/aV7oTuI+WoLQZVEDXg=",
            },
            {
                "account": "NMA2FKN8up2cEwaJgtmAiDrZWB69ApnDfp",
                "scopes": "CalledByEntry",
            }
        ]
    ],
    "id": 1
}
```


And then parse the provided witnesses in `GetSignersFromJson method (https://github.com/neo-project/neo-modules/blob/master/src/RpcServer/RpcServer.SmartContract.cs#L94).

And then fill the transaction like the following:
```
            Transaction tx = new Transaction
            {
                Witnesses = signers.Witnesses,
                ...
            }
```

Reply:
It will help us to get rid of the private key dependency (because if we parse witnesses, then we don't need private keys in the RPC node wallet anymore).

Reply:
Ok